### ISSUE TITLE
Automatischer Versand/Export

### ISSUE THREAD

**Issue by sheepyhollow** [mimischi/django-clock#406]
Alternativ zum Download einer PDF-Datei kann der fertige Stundenzettel auch per Mail versandt werden.
Aber welche Absenderadresse? Eine Zuordnung zum Benutzer wäre schon gut - oder müsste im Betreff stehen.

Ggf. die Idee im Kopf behalten, einen Upload auf eine vorzugebende Adresse (FTP o.ä.) zu ermöglichen.

**Comment by mimischi** [comment]
Insgesamt macht das natürlich eine größere Problematik auf. Aktuell werden die Stundenzettel On-Demand generiert und gar nicht zwischengespeichert. Man könnte die Stundenzettel natürlich generieren und auf dem Dateisystem abspeichern. Der Abruf würde dann einfach den Download (und nicht, wie aktuell, das erneute generieren) anstoßen. Ändere ich dann einen vergangenen Monat ab, so wird die Datei neu generiert.

> Alternativ zum Download einer PDF-Datei kann der fertige Stundenzettel auch per Mail versandt werden.
> Aber welche Absenderadresse? Eine Zuordnung zum Benutzer wäre schon gut - oder müsste im Betreff stehen.

Soll das an den Vorgesetzten oder an den HiWi gehen? Was passiert, wenn es automatisiert an den Vorgesetzten geht, der HiWi aber noch nicht alle Stunden eingetragen hat? Dann kriegt man ja erstmal einen unvollständigen Stundenzettel.
Sobald der HiWi die Stunden nachträgt, geht dann erneut eine Mail automatisch raus mit dem Zettel? Oder passiert das dann manuell?


Soll ein HiWi überhaupt nachträglich Arbeitszeiten ändern können?
Beispiel: Am 01.10.2017 kriegt der Vorgesetzte den Stundenzettel automatisch geschickt. Der HiWi hat nun keine Möglichkeit mehr am September Änderungen durchzuführen, weil der Stundenzettel schon generiert wurde.

Andererseits ist das wieder eine Einschränkung der Nutzung - einen ausgedruckten Zettel kann ich ja jeder Zeit neu ausfüllen.

**Comment by sheepyhollow** [comment]
Nach Vorschlag von Herrn Scharf wird der Zettel idealerweise "nach Abschluss des letzten Dienstes" abgegeben. Danach ist das quasi in Stein gemeißelt. Und ich will um Gottes Willen nur eine Version pro Hiwi und Monat! :) 
Es soll auch kein automatischer Export sein - insofern ist der issue-Titel irreführend (umbenannt).
Ich dachte nur daran, dass man an Stelle des Downloads einer Datei (die ich dann wieder in eine Mail stecken muss etc.) diese direkt aus Clock heraus versenden kann. Also nicht automatisch, sondern aktiv und vom Benutzer getriggert und mit so vielen "sind sie sicher!?" wie möglich.

**Comment by mimischi** [comment]
Machbar wäre es definitiv. Man kann dem Nutzer auch die Möglichkeit geben einen _Review_ des Stundenzettels zu machen, bevor dieser verschickt wird.

Der Betreff der Mail wäre auch anpassbar. Wenn man das geschickt macht, könnte man eben "Vorgesetzten"-Accounts definieren. Entweder weißt sich dann jeder HiWi einen Vorgesetzten zu oder umgekehrt. Der Versand erfolgt dann an die Kontaktadresse.

**Comment by sheepyhollow** [comment]
Ja, genau so in der Art. Es muss alles in der Kontrolle des Nutzers bleiben.
